frontend/server/src/main/java/com/amazonaws/ml/mms/util/logging/QLogLayout.java

```diff
@@ -32,9 +32,10 @@ public String format(LoggingEvent event) {
             if (metric.getRequestId() != null) {
                 stringBuilder.append("\nRequestId=").append(metric.getRequestId());
             }
-            stringBuilder.append("\nTimeStamp=").append(metric.getTimestamp());
+            stringBuilder.append("\nStartTime=").append(metric.getTimestamp());
```

Comment:
I think the required fields are "StartTime", "Time", "ProgramName" and "MarketPlace". We should probably replace "EndTime" with "Time" where "Time = StartTime + Time" and MarketPlace, we will need to see how to provide it.

Reply:
Yes, I agree with Vamshi. Also, please note that unit for StartTime is *second* while unit for Time is *millisecond*. I think Time could be just EndTime (as currently available) - (minus) StartTime as currently available.

As for Marketplace, that is very interesting question and I don't think we can solve it quickly as it will require to introduce some customary attributes into logs that at passed all the way from configs. We need to file a backlog issue for it and not address here.